### ISSUE TITLE
[Feature] [Performance] Improve X11 screenshot performance

### ISSUE THREAD

**Issue by aimmac23** [aimmac23/selenium-video-node#14]
Most of the CPU time taken to record a video seems is taken up by time spent in the java.awt.Robot class. Having read the JVM native source code for the X11 version, it didn't strike me as very efficient - lots of looping and copying.

Adding some native code might be faster:

http://stackoverflow.com/questions/6063329/screenshot-using-opengl-and-or-x11

Only useful for people running on Linux, since OS X and Windows doesn't use X11.

**Comment by aimmac23** [comment]
Relevant Java source code:

https://github.com/openjdk-mirror/jdk7u-jdk/blob/f4d80957e89a19a29bb9f9807d2a28351ed7f7df/src/solaris/native/sun/awt/awt_Robot.c#L201

**Comment by aimmac23** [comment]
This has now been implemented, although not by default. The improved performance mode can be enabled by passing "-Dvideo.source=x11" to the JVM. I've seen performance go up by at least 4x on my laptop, but other people's mileage may vary.

I spotted a few minor visual artefacts when recording using this mode (only saw it on Firefox's search bar so far),so its possible that this affects the rendering of the screen a bit (and depending on the hardware/drivers as well).

Let's see how this goes a bit longer before making it enabled by default.
